### DISCUSSION TITLE
Exposing more page data for reference on other pages?

### DISCUSSION THREAD

**Discussion by rowsof** [zim-desktop-wiki/zim-desktop-wiki#2443] [Ideas]
I love it that I can name a page something, link to that page all over the place, and later decide to rename the page and have that rename propagate all over my wiki. There are two aspects of this that I find empowering:

- The links still work.
- The text representing the link changes throughout the wiki, allowing that information to remain "true" everywhere it's referenced.

I think it'd be super valuable to be able to do this for other data.

- I'd love to be able to do this for headings within a page, as I find myself linking more and more often to different headings but worry that I'll definitely edit that heading at some point and break the link (and the changed data won't propagate either).
- This next part I think has already been requested, and likely has been deemed outside of what zim can do + way out of scope, but I also think it'd be amazing to be able to do this for larger blocks of text; and wonder if there's an approach similar to the page "Rename" approach that would make this possible. It'd be so cool to be able to select a chunk of text on a page, right-click, and mark it as a quotable/accessible block, and then to be able to paste that block elsewhere. It could basically function as a piece of linked text with custom formatting to distinguish it from standard links and communicate that it is un-editable in the quote location, and in order to change the text in its referenced location you'd need to do an action similar to the right-click > rename functionality that's currently available for pages.

Any thoughts? Cheers!

**Comment by introt** [comment]
Related to the first bullet: #1759

For the latter part see links here: https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1818#discussioncomment-1815552
